pkg/urlutil/urlutil_test.go

```diff
@@ -6,17 +6,21 @@ var (
 	gitUrls = []string{
 		"git://github.com/docker/docker",
 		"git@github.com:docker/docker.git",
-		"git@bitbucket.org:atlassianlabs/atlassian-docker.git",
+		"git@bitbucket.org:atlassianlabs/atlassian-docker",
 		"https://github.com/docker/docker.git",
 		"http://github.com/docker/docker.git",
 		"http://github.com/docker/docker.git#branch",
 		"http://github.com/docker/docker.git#:dir",
+		"git+ssh://github.com/docker/docker",
+		"git+https://github.com/docker/docker.git#branch",
+		"git+ftps://github.com/docker/docker.git#:dir",
```

Comment:
I'd rather not rely on what's _possibly_ supported by the (version of) git that's installed; I suggest to limit to `ssh`, `https`, and we can add more if desired at some point